src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPathExtensions.cs

```diff
@@ -21,7 +21,7 @@ namespace Microsoft.OData.UriParser
     /// <remarks>
     /// The values that these methods compute are not cached.
     /// </remarks>
-    internal static class ODataPathExtensions

```

Comment:
We can't implement the required logic in Web API side?

Reply:
Some of the classes used by the extension methods e.g `PathSegmentToResourcePathTranslator` are also internal in ODL.